src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+     *                                   and the second is another tuple of the temporary directory and a temporary FASTQ file
+     *                                   containing sequence information.
+     * @throws IOException               an IOException is thrown if the conversion fails to create the FASTQ file
+     */
+    @VisibleForTesting
+    static Tuple2<Long, File> convertReadsToFASTQ(final Tuple2<Long, Iterable<GATKRead>> orderedReadsOfABreakPoint) throws IOException{
+
+        final Long breakpointId = orderedReadsOfABreakPoint._1();
+
+        File workingDir = new File( Files.createTempDirectory("assembly" + orderedReadsOfABreakPoint._1().toString()).toString() );
+        if(null==workingDir){
+            throw new IOException("Failed to create temporary directory to perform assembly in.");
+        }
+        IOUtils.deleteRecursivelyOnExit(workingDir);
+
+        final File fastq = new File(workingDir, "assembly" + breakpointId + ".fastq");
+        if(null==fastq){
+            throw new IOException("Failed to create temporary FASTQ file to feed to SGA.");
+        }
+
+        try(final FastqWriter writer = new FastqWriterFactory().newWriter(fastq)){
+            Iterator<GATKRead> it = orderedReadsOfABreakPoint._2().iterator();
+            while(it.hasNext()){
+                writeToTempFASTQ(it.next(), writer);
+                writeToTempFASTQ(it.next(), writer);
+            }
+        }
+
+        return new Tuple2<>(breakpointId, fastq);
+    }
+
+    // copied from FindSVBreakpointsSpark
+    private static void writeToTempFASTQ(final GATKRead read, final FastqWriter writer){
+        final String readName      = read.getName();// TODO: hack for unit testing, where read names have /1 and /2. should be:   + (read.isSecondOfPair() ? "/2" : "/1");
+        final String readBases     = read.getBasesString();
+        final String baseQualities = ReadUtils.getBaseQualityString(read);
+        writer.write(new FastqRecord(readName, readBases, "", baseQualities));
+    }
+
+    /**
+     * Final return type of the whole process of SGA local assembly.
+     * assembledContigFile is the file containing the assembled contigs, if the process executed successfully, or null if not.
+     * runtimeInformation contains the runtime information logged along the process up until the process erred, if errors happen,
+     *  or until the last step if no errors occur along the line.
+     *  The list is organized along the process of executing the assembly pipeline, inside each entry of the list is
+     *      the name of the module,
+     *      return status of the module,
+     *      stdout message of the module, if any,
+     *      stderr message of the module, if any
+     */
+    @VisibleForTesting
+    static final class SGAAssemblyResultType implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        public final File assembledContigFile;
```

Comment:
I see, you do do read in the file eventually and turn it into a ContigsCollection. Since this is going into an RDD, it might be better to parse the result file if it exists immediately into a ContigsCollection, say in the constructor for this class.